substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/management/ManagementSupport.java

```diff
@@ -173,6 +179,16 @@ private synchronized OperatingSystemMXBean getOsMXBean() {
         return osMXBean;
     }
 
+    private synchronized FlightRecorderMXBean getFlightRecorderMXBean() {
+        if (!HasJfrSupport.get()) {
```

Comment:
Should this check whether JMX is included/enabled? It seems that `Target_jdk_management_jfr_FlightRecorderMXBeanImpl` pulls in a lot of JMX code.

Reply:
I'm taking care of this...

Reply:
hmm the reason it doesn't depend on whether the `jmxserver` option is enabled is because it could still be needed in the mbean server for local JMX use (even if remote JMX use is disabled).

Reply:
That being said, maybe it's not worth the cost of including `FlightRecorderMXBean` in the mbean server unless `jmxserver` is enabled. Because normal JFR can be used locally instead of the `FlightRecorderMXBean`.